syntax = "proto3";
package com.fiberhome.fums.proto;
option java_package = "com.fiberhome.fums.proto";   
option java_outer_classname = "App";
import "CommonDefine.proto";

message AppData
{
	enum APP_OP_TYPE
	{
		ADD    = 0;
		UPDATE = 1;
		DEL    = 2;
	}
	APP_OP_TYPE  op_type                                 = 1;
	string appType                                       = 2;
	EnumDefineData.AppTypeEnum appTypeCode               = 3;
	string appVersion                                    = 4;
	
	enum AppStatusTypeEnum
	{
		STOPPED    = 0;
		RUNNING    = 1;
	}
	
	AppStatusTypeEnum  appStatus                        = 5;
	string appTryInstallingVersion                      = 6;
	
	enum AppInstallStatusEnum
	{
		INSTALLED_SUCCESS          = 0;
		INSTALLED_FAILED           = 1;
		DOWNLOAD_SUCC_NOT_INSTALL  = 2;
		DOWNLOAD_FAILED            = 3;
	}
	
	AppInstallStatusEnum  appTryInstallingStatus       = 7;
    string appTryInstallingFailedReason                = 8;	
	bool   isMayStart                                  = 9;
	int32  startCount                                  = 10;
	int32  appDemonStatus                              = 11;	
}

